src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metricsdb/MetricsDB.java

```diff
@@ -259,6 +308,23 @@ public void deleteOnDiskFile() {
     }
   }
 
+  /**
+   * Deletes the metricsdb file associated with the given timestamp if it exists.
+   *
+   * @param windowStartTime the timestamp associated with an existing metricsdb file
+   */
+  public static void deleteOnDiskFile(long windowStartTime) {
+    String dbFilePath = getDBFilePath(windowStartTime);
+    File dbFile = new File(dbFilePath);
```

Comment:
Such a check for the existence of the file is unnecessary -- perhaps after the check the file may not be there. That is why the delete operation itself takes care of all of that.
We should keep it's name as "deleteOnDiskFile" to stay consistent with current "deleteOnDiskFile". Both files to do the same thing, it's just this new version is static and accepts a windowStartTime argument.

Reply:
In that case use Files.delete() and catch the exception if one is thrown  for understanding why it could not be deleted. The error message here is not helpful

Reply:
Sure

Reply:
> it's just this new version is static and accepts a windowStartTime argument.

why do we need a separate static instance of the same method, instead of overloading `public void deleteOnDiskFile()` ?

Reply:
There's two separate abstractions we want. 1. Given a MetricsDB instance, we want to be able to delete it's on-disk file (the non-static version). 2. We want to be able to able to delete arbitrary metrcsdb files without creating an instance (the static version). Creating an instance would have the cost of setting up a sql connection to it... which itself could actually fail (that's why the ctor throws an Exception).